pkg/graph/resolvers/model_plan.go

```diff
@@ -537,3 +537,40 @@ func ModelPlanShare(
 
 	return true, nil
 }
+
+// ModelPlanAnticipatedPhase calculates a suggested phase for a model plan based on its current status and timeline
+// It uses a series of status evaluation strategies to determine the suggested phase
+// If no phase is suggested, it returns nil
+func ModelPlanAnticipatedPhase(
+	ctx context.Context,
+	logger *zap.Logger,
+	store *storage.Store,
+	modelPlanID uuid.UUID,
+) (*model.PhaseSuggestion, error) {
+
+	modelPlan, err := store.ModelPlanGetByID(store, logger, modelPlanID)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the model plan is paused or canceled, we shouldn't suggest a new phase
+	if modelPlan.Status == models.ModelStatusPaused || modelPlan.Status == models.ModelStatusCanceled {
+		return nil, nil
+	}
+
+	planBasics, err := PlanBasicsGetByModelPlanIDLOADER(ctx, modelPlanID)
```

Comment:
@OddTomBrooks , I think this flow could be revisited here. I think the dataloader is good and should be used for the model plan resolver where there is a possibility of an N+1 scenario. However, that won't work in the data loader. 

Ideally, the resolver should use the data loader, and the job should call the store. This might just take some refactoring (perhaps splitting out the analysis logic from the data fetching logic?